pkg/dogstatsd/parser.go

```diff
@@ -62,7 +63,9 @@ func parseTags(rawTags []byte, extractHost bool) ([]string, string) {
 	if len(rawTags) == 0 {
 		return nil, ""
 	}
-	var host string
+
+	// GetHostname should be cached
+	host, _ := util.GetHostname()
```

Comment:
even though it's supposed to be cached, I'd be more reassured (perf-wise) if we only fetched it once in the lifetime of dogstatsd 😇

Reply:
also, note that doing this logic here will make the change apply to events and service checks too. I think Agent 5 does that too (better if you check, but if it's the case, great), but let's mention it in the release note

Reply:
Good catch, Agent5 only do this for metrics, not for event or service_check.

Reply:
Actually the hostname for service_check and event is not deduce from tags, so we're good.